pkg/secret/provider/context_provider.go

```diff
@@ -0,0 +1,353 @@
+package provider
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"code.google.com/p/go-uuid/uuid"
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet"
+	"github.com/fsouza/go-dockerclient"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	secretapi "github.com/openshift/origin/pkg/secret/api"
+)
+
+const (
+	SecretDomain    = "secret.openshift.com"
+	EnvSecretType   = "ENV"
+	FileSecretType  = "FILE"
+	RequiredMarker  = "!"
+	DefaultFileMode = os.FileMode(0644)
+)
+
+func NewSecretProvider(rootDir string, osClient osclient.Interface, kClient kclient.Interface) kubelet.SecurityContextProvider {
+	return &secretProvider{
+		rootDir:  rootDir,
+		osClient: osClient,
+		kClient:  kClient,
+	}
+}
+
+type secretProvider struct {
+	rootDir  string
+	osClient osclient.Interface
+	kClient  kclient.Interface
+}
+
+type envSecret struct {
+	name     string
+	required bool
+}
+
+type fileSecret struct {
+	path     string
+	owner    int
+	group    int
+	mode     os.FileMode
+	required bool
+}
+
+func (p *secretProvider) getPath(pod *kapi.BoundPod, container *kapi.Container, secretName string) string {
+	return filepath.Join(p.rootDir, pod.Namespace, pod.Name, "secrets", container.Name, "volumes", secretName)
+}
+
+func (p *secretProvider) SecureContainer(
+	boundPod *kapi.BoundPod,
+	container *kapi.Container,
+	createOptions *docker.CreateContainerOptions) error {
+
+	pod, err := p.kClient.Pods(boundPod.Namespace).Get(boundPod.Name)
+	if err != nil {
+		return err
+	}
+	envSecrets := p.getContainerEnvSecrets(pod, container.Name)
+	if len(envSecrets) > 0 {
+		envs, err := p.provisionEnvSecrets(pod.Namespace, envSecrets)
+		if err != nil {
+			return err
+		}
+		createOptions.Config.Env = append(createOptions.Config.Env, envs...)
+	}
+	return nil
+}
+
+func (p *secretProvider) SecureHostConfig(
+	boundPod *kapi.BoundPod,
+	container *kapi.Container,
+	hostConfig *docker.HostConfig) error {
+
+	pod, err := p.kClient.Pods(boundPod.Namespace).Get(boundPod.Name)
+	if err != nil {
+		return err
+	}
+	fileSecrets := p.getContainerFileSecrets(pod, container.Name)
+	if len(fileSecrets) > 0 {
+		dir, err := p.setupSecretsDir(pod.Namespace, pod.Name, container.Name)
+		binds, err := p.provisionFileSecrets(pod.Namespace, dir, fileSecrets)
+		if err != nil {
+			return err
+		}
+		hostConfig.Binds = append(hostConfig.Binds, binds...)
+	}
+	return nil
+}
+
+// getContainerEnvSecrets returns a map of secret descriptors to be passed as environment variables to the
+// container to be created
+// The pod is expected to contain annotations in the format of:
+// secret.openshift.com/[secretName] = ENV:<containerName>:<EnvVarName>[:!]
```

Comment:
It's a namespace for the annotation. https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/types.go#L110-L126